src/backend/distributed/commands/function.c

```diff
@@ -146,6 +146,17 @@ create_distributed_function(PG_FUNCTION_ARGS)
 
 	ObjectAddressSet(functionAddress, ProcedureRelationId, funcOid);
 
+	if (IsObjectAddressOwnedByExtension(&functionAddress))
+	{
+		char *functionName = get_func_name(funcOid);
+		ereport(ERROR, (errmsg("unable to create a distributed function from functions "
```

Comment:
can we find the extension name as well? Printing that would be helpful in the error messages.

Also, super minor, to keep this function smaller and avoid verbose error lines, it'd have a slight preference to move this check to `ErrorIfFunctionDependsOnExtension()`. But, feel free to ignore this if you don't feel similar.